src/palace.pyx

```diff
@@ -1701,7 +1725,133 @@ cdef class AuxiliaryEffectSlot:
+        """The effect with the specified reverb properties.
+        It will automatically downgrade to the Standard Reverb effect
+        if EAXReverb effect is not supported.
+        """
+        cdef EFXEAXREVERBPROPERTIES properties
+        try:
+            properties.flDensity = value['density']
+            properties.flDiffusion = value['diffusion']
+            properties.flGain = value['gain']
+            properties.flGainHF = value['gain_hf']
+            properties.flGainLF = value['gain_lf']
+            properties.flDecayTime = value['decay_time']
+            properties.flDecayHFRatio = value['decay_hf_ratio']
+            properties.flDecayLFRatio = value['decay_lf_ratio']
+            properties.flReflectionsGain = value['reflections_gain']
+            properties.flReflectionsDelay = value['reflections_delay']
+            properties.flReflectionsPan[0] = value['reflections_pan'][0]
+            properties.flReflectionsPan[1] = value['reflections_pan'][1]
+            properties.flReflectionsPan[2] = value['reflections_pan'][2]
+            properties.flLateReverbGain = value['late_reverb_gain']
+            properties.flLateReverbDelay = value['late_reverb_delay']
+            properties.flLateReverbPan[0] = value['late_reverb_pan'][0]
+            properties.flLateReverbPan[1] = value['late_reverb_pan'][1]
+            properties.flLateReverbPan[2] = value['late_reverb_pan'][2]
+            properties.flEchoTime = value['echo_time']
+            properties.flEchoDepth = value['echo_depth']
+            properties.flModulationTime = value['modulation_time']
+            properties.flModulationDepth = value['modulation_depth']
+            properties.flAirAbsorptionGainHF = value['air_absorption_gain_hf']
+            properties.flHFReference = value['hf_reference']
+            properties.flLFReference = value['lf_reference']
+            properties.flRoomRolloffFactor = value['room_rolloff_factor']
+            properties.iDecayHFLimit = value['decay_hf_limit']
+        except KeyError:
+            pass
+        self.impl.set_reverb_properties(properties)
+        
+    @setter
+    def chorus_properties(self, value: dict) -> None:
+        """The effect with the specified chorus properties.
+        It will be thrown if EAXReverb effect is not supported.
+        """
+        cdef EFXCHORUSPROPERTIES properties
+        try:
+            properties.iWaveform = value['waveform']
+            properties.iPhase = value['phase']
+            properties.flRate = value['rate']
+            properties.flDepth = value['depth']
+            properties.flFeedback = value['feedback']
+            properties.flDelay = value['delay']
+        except KeyError:
+            pass
+        self.impl.set_chorus_properties(properties)
+
+    def destroy(self) -> None:
+        """Destroy the effect, removing all effects from it
+        before being freed.
+        """
+        self.impl.destroy()
 
+
```

Comment:
What is this?

Reply:
A newline? I guess?

Reply:
I think it should be a form feed. Also check if you have any space there because extra space is dangerous in python.

Reply:
Not really but people in general hates trailing whitespace.  Trust me I'm a person.

Reply:
idk but seems like vscode doesnt show the form feed :/

Reply:
nvm that, now how can I insert formfeed

Reply:
You messed up the formfeed's position.